stellar-contract-env-host/src/host.rs

```diff
@@ -751,18 +768,33 @@ impl Host {
         Ok(id_obj)
     }
 
+    fn try_call_vtable(
+        &self,
+        id: &Hash,
+        func: &Symbol,
+        args: &[RawVal],
+    ) -> Option<Result<RawVal, HostError>> {
+        #[cfg(feature = "testutils")]
+        if let Some(vtable) = self.0.vtables.borrow().get(&id) {
+            if let Some(f) = vtable.0.get(&func) {
+                let mut frame_guard = self.push_test_frame(id.clone());
+                let res = Some(Ok(f.0.call(args)));
+                frame_guard.commit();
+                res
+            } else {
+                Some(Err(HostError::General("function not in vtable")))
+            }
+        } else {
+            None
+        }
+
+        #[cfg(not(feature = "testutils"))]
+        None
+    }
+
     #[cfg(feature = "vm")]
-    fn call_n(&self, contract: Object, func: Symbol, args: &[RawVal]) -> Result<RawVal, HostError> {
+    fn call_vm(&self, id: &Hash, func: &Symbol, args: &[RawVal]) -> Result<RawVal, HostError> {
```

Comment:
I renamed this `call_wasm_fn` to make it more obvious what it actually does. It's just a helper function for `call_n` to avoid it being ridiculously long.